router/throttler/adaptiveRateLimit.go

```diff
@@ -0,0 +1,112 @@
+package throttler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+)
+
+type timers struct {
+	timer        time.Duration
+	limitReached map[string]bool
+	mu           sync.Mutex
+	limitSet     bool
+}
+
+func SetupRouterAdaptiveRateLimiter(ctx context.Context, errorCh <-chan string) func(destName, destID string, limit int64) int64 {
+	shortTimeFrequency := config.GetDuration("Router.throttler.adaptiveRateLimit.shortTimeFrequency", 5, time.Second)
```

Comment:
Let's DI  and use  `*config.Config` instead of using the default one. It helps with isolation during testing.